Git 2.19.0

Mno-hime opened the pull request:
* Git now defaults to pcre2 library
* removed Apache test configuration as they weren't triggered anyway
* Perl patch not needed anymore
* removed GNU path as it stucks tests
* new failed test: t7005

Tested: `log`, `show`, `status`, `add`, `rm`, `mv`, `commit`, `push`,
and `rebase` commands. Comming signing works as well.

Mno-hime committed:
Git 2.19.0

* Git now defaults to pcre2 library
* removed Apache test configuration as they weren't triggered anyway
* Perl patch not needed anymore
* removed GNU path as it stucks tests
* new failed test: t7005

Tested: `log`, `show`, `status`, `add`, `rm`, `mv`, `commit`, `push`,
and `rebase` commands. Comming signing works as well.

pyhalov commented:
There's an issue with new t7005 test. It creates file ```'echo space >$1'``` , but should ```'echo space > "$1"'```:
```
$ cat patches/0005-t7005-editor.sh.patch 
--- git-2.19.0/t/t7005-editor.sh.~1~    Mon Sep 10 21:39:13 2018
+++ git-2.19.0/t/t7005-editor.sh        Wed Sep 26 09:41:30 2018
@@ -112,7 +112,7 @@
 done
 
 test_expect_success 'editor with a space' '
-       echo "echo space >\$1" >"e space.sh" &&
+       echo "echo space > \"\$1\"" > "e space.sh" &&
        chmod a+x "e space.sh" &&
        GIT_EDITOR="./e\ space.sh" git commit --amend &&
        test space = "$(git show -s --pretty=format:%s)"
```
Please, incorporate the patch and re-test.

Mno-hime commented:
Thanks, @pyhalov. I incorporated the patch, updated the test master file.

There's instability in `t0202-gettext-perl`'s sub-test's execution order:
```
@@ -2902,9 +2902,9 @@
 ok 2 - Git::I18N is located at $(@D)/t/../perl/build/lib/Git/I18N.pm
 ok 3 - sanity: Git::I18N has 3 export(s)
 ok 4 - sanity: Git::I18N exports everything by default
-ok 5 - sanity: __ has a $ prototype
-ok 6 - sanity: __n has a $$$ prototype
-ok 7 - sanity: N__ has a $ prototype
+ok 5 - sanity: N__ has a $ prototype
+ok 6 - sanity: __ has a $ prototype
+ok 7 - sanity: __n has a $$$ prototype
 ok 8 - Passing a string through __() in the C locale works
 ok 9 - Get singular string through __n() in C locale
 ok 10 - Get plural string through __n() in C locale
```
Probably not worth fixing.
